packages/core/src/instrumentation.ts

```diff
@@ -0,0 +1,46 @@
+import { AsyncLocalStorage } from "async_hooks";
+import { inspect } from "util";
+import { v4 as uuidv4 } from "uuid";
+
+export type ITrace = (run: string | null, time: Date, name: string, data: any) => void;
+
+export function generateRunUUID(): string {
+    return uuidv4();
+}
+
+export class Instrumentation {
+    private static store = new AsyncLocalStorage<Instrumentation>();
+    private static tracer: ITrace = null;
+    private run: string = null;
+
+    static init(tracer: ITrace) {
+        this.tracer = tracer;
+    }
+
+    static trace(name: string, data: any) {
+        if (!this.tracer) return;
+        var jsonPayload;
+        try {
+            jsonPayload = JSON.stringify(data);
+        } catch (error) {
+            name = `ERROR: ${name}`;
+            jsonPayload = {ok: false, name: name, error: `${error}, inspect: ${inspect(data)}`};
+            console.log(`Could not convert object to JSON for ${name}. Please select individual fields that are serializable.`);
```

Comment:
eliza log everywhere

Reply:
This is intentional because (1) I don't want to see the colors, (2) I need the raw printout, and (3) this simply does not belong in the same category as logging.

The purpose for this is to catch the objects that fail to format, and then selectively pick the properties we need.